packages/yoroi-extension/app/api/ada/transactions/shelley/transactions.js

```diff
@@ -695,7 +697,30 @@ function _newAdaUnsignedTxFromUtxo(
         const isRemainingNeededCoinZero = isBigNumZero(remainingNeeded.coin());
         const isRemainingNeededAssetZero = (remainingAssets?.len() ?? 0) === 0;
         if (isRemainingNeededCoinZero && isRemainingNeededAssetZero && isNonEmptyInputs) {
-          if (oneExtraInput) {
+          const changeTokenIdSet = new Set(
+            parseTokenList(excessiveInputAssets).map(r => r.assetId)
+          );
+          let packed = false;
+          for (let j = i; j < utxos.length; j++) {
+            const packCandidate = utxos[j];
+            if (
+              packCandidate.assets.length >= 1 &&
+                packCandidate.assets.every(({ assetId }) => changeTokenIdSet.has(assetId))
+            ) {
+              addUtxoInput(
+                txBuilder,
+                undefined,
+                packCandidate,
+                false,
+                { networkId: protocolParams.networkId }
+              );
+              usedUtxos.push(packCandidate);
+
+              packed = true;
+              break;
+            }
+          }
+          if (oneExtraInput && !packed) {
```

Comment:
packing an additional token input should not affect the "extra-input" logic, because the branch with `oneExtraInput = false` will also pack the additional one for token - so this breaks the logic.

```suggestion
          if (oneExtraInput) {
```

Reply:
But when we *do* pack an input, there is no need for an extra input, because the packed input serves as the "extra" one.

Reply:
Ah, because if we found an additional input where all assets are already present in the tx - we definitely have enough ADA purified from that utxo to cover getting the change. Good point, I missed that.